### PR TITLE
Generate Auditbeat module config from template

### PR DESCRIPTION
**Pull request by andrewkroh**
Use a template to generate reference and non-reference configs from the same template.
The template is also used to generate OS specific configs for use in packages.

### CODE REVIEW

File: auditbeat/module/audit/_meta/config.yml.tpl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    binaries:
+    - /bin
+    - /usr/bin
+    - /usr/local/bin
+    - /sbin
+    - /usr/sbin
+    - /usr/local/sbin
+  {{ else if eq .goos "windows" -}}
+  file.paths:
+    windows:
```
Comment (tsg) [inline auditbeat/module/audit/_meta/config.yml.tpl]: I didn't realize the file integrity module supports Windows. Cool 👍

File: auditbeat/auditbeat.reference.yml [hunk 2]
```diff
@@ -26,38 +26,43 @@ auditbeat.config.modules:
...
-  # Kernel Metricset Options
+  metricsets: [kernel]
   kernel.resolve_ids: true
   kernel.failure_mode: silent
   kernel.backlog_limit: 8196
   kernel.rate_limit: 0
   kernel.include_raw_message: false
   kernel.include_warnings: false
-  kernel.audit_rules: |
+  kernel.audit_rules:
```
Comment (tsg) [inline auditbeat/auditbeat.reference.yml]: Is the removal of `|` intentional?

Comment (andrewkroh) [inline auditbeat/auditbeat.reference.yml]: Unintentional. Will fix.

Comment (andrewkroh) [inline auditbeat/auditbeat.reference.yml]: Fixed (and rebased + squashed).

**Comment by andrewkroh** [comment]
The build failure looks unrelated to my changes (it was a failure to wipe the filebeat workspace).

**Pull request merged**